File: coyote/apis/kv.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    KvDeleteIn,
    KvDeleteOut,
    KvGetIn,
    KvGetOut,
    KvSetIn,
    KvSetOut,
)
from .kv_namespace import (
    KvNamespace,
    KvNamespaceAsync,
)

from ..models.kv_set_in import _KvSetIn
from ..models.kv_get_in import _KvGetIn
from ..models.kv_delete_in import _KvDeleteIn


class KvAsync(ApiBase):
    @property
    def namespace(self) -> KvNamespaceAsync:
        return KvNamespaceAsync(self._client)

    async def set(
        self,
        key: str,
        kv_set_in: KvSetIn,
    ) -> KvSetOut:
        """KV Set"""
        body = _KvSetIn(
            namespace=kv_set_in.namespace,
            key=key,
            value=kv_set_in.value,
            ttl=kv_set_in.ttl,
            behavior=kv_set_in.behavior,
            version=kv_set_in.version,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.kv.set",
            body=body,
            response_type=KvSetOut,
        )

    async def get(
        self,
        key: str,
        kv_get_in: KvGetIn = KvGetIn(),
    ) -> KvGetOut:
        """KV Get"""
        body = _KvGetIn(
            namespace=kv_get_in.namespace,
            key=key,
            consistency=kv_get_in.consistency,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.kv.get",
            body=body,
            response_type=KvGetOut,
        )

    async def delete(
        self,
        key: str,
        kv_delete_in: KvDeleteIn = KvDeleteIn(),
    ) -> KvDeleteOut:
        """KV Delete"""
        body = _KvDeleteIn(
            namespace=kv_delete_in.namespace,
            key=key,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.kv.delete",
            body=body,
            response_type=KvDeleteOut,
        )


class Kv(ApiBase):
    @property
    def namespace(self) -> KvNamespace:
        return KvNamespace(self._client)

    def set(
        self,
        key: str,
        kv_set_in: KvSetIn,
    ) -> KvSetOut:
        """KV Set"""
        body = _KvSetIn(
            namespace=kv_set_in.namespace,
            key=key,
            value=kv_set_in.value,
            ttl=kv_set_in.ttl,
            behavior=kv_set_in.behavior,
            version=kv_set_in.version,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.kv.set",
            body=body,
            response_type=KvSetOut,
        )

    def get(
        self,
        key: str,
        kv_get_in: KvGetIn = KvGetIn(),
    ) -> KvGetOut:
        """KV Get"""
        body = _KvGetIn(
            namespace=kv_get_in.namespace,
            key=key,
            consistency=kv_get_in.consistency,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.kv.get",
            body=body,
            response_type=KvGetOut,
        )

    def delete(
        self,
        key: str,
        kv_delete_in: KvDeleteIn = KvDeleteIn(),
    ) -> KvDeleteOut:
        """KV Delete"""
        body = _KvDeleteIn(
            namespace=kv_delete_in.namespace,
            key=key,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.kv.delete",
            body=body,
            response_type=KvDeleteOut,
        )

File: src/api/kv.rs
// this file is @generated
use super::KvNamespace;
use crate::{Configuration, error::Result, models::*};

pub struct Kv<'a> {
    cfg: &'a Configuration,
}

impl<'a> Kv<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn namespace(&self) -> KvNamespace<'a> {
        KvNamespace::new(self.cfg)
    }

    /// KV Set
    pub async fn set(&self, key: String, kv_set_in: KvSetIn) -> Result<KvSetOut> {
        let kv_set_in = KvSetIn_ {
            namespace: kv_set_in.namespace,
            key,
            value: kv_set_in.value,
            ttl: kv_set_in.ttl,
            behavior: kv_set_in.behavior,
            version: kv_set_in.version,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.kv.set")
            .with_body(kv_set_in)
            .execute(self.cfg)
            .await
    }

    /// KV Get
    pub async fn get(&self, key: String, kv_get_in: KvGetIn) -> Result<KvGetOut> {
        let kv_get_in = KvGetIn_ {
            namespace: kv_get_in.namespace,
            key,
            consistency: kv_get_in.consistency,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.kv.get")
            .with_body(kv_get_in)
            .execute(self.cfg)
            .await
    }

    /// KV Delete
    pub async fn delete(&self, key: String, kv_delete_in: KvDeleteIn) -> Result<KvDeleteOut> {
        let kv_delete_in = KvDeleteIn_ {
            namespace: kv_delete_in.namespace,
            key,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.kv.delete")
            .with_body(kv_delete_in)
            .execute(self.cfg)
            .await
    }
}
